File: 2023/01/t.py
cal = open("input.txt", "r", encoding="utf-8").read().split("\n")

def ret(x):
    nums = "".join(i for i in x if i.isdigit())
    return int(nums[0]) * 10 + int(nums[-1])

def subs(x):
    for i, name in enumerate(["one", "two", "three", "four", "five", "six", "seven", "eight", "nine"]):
        # It's necessary to concatenate the first and last letter of the name because some lines have, for example: "oneight" (2 names joined)
        x = x.replace(name, name[0] + str(i + 1) + name[-1:]) # the first letter of the name + number + the last letter of the name
    return ret(x)


print("Round 1:", sum(map(ret, cal))) # 55386

print("Round 2:", sum(map(subs, cal)))  # 54824

File: 2023/01/t.cs
var lines = File.ReadAllLines("input.txt");
var counter1 = 0;
var counter2 = 0;

foreach (var line in lines)
{
    int RetNum(string s)
    {
        return (s.First(Char.IsDigit) - '0') * 10 + s.Last(Char.IsDigit)  - '0';
    }
    string ReplaceNums(string s)
    {
        string[] nums = { "one", "two", "three", "four", "five", "six", "seven", "eight", "nine" };
        for(int x = 1; x <= nums.Length; x++)
        {
            s = s.Replace(nums[x - 1], nums[x - 1][0].ToString() + x.ToString() + nums[x - 1][nums[x - 1].Length - 1].ToString());
        }
        return s;
    }


    counter1 += RetNum(line);
    counter2 += RetNum(ReplaceNums(line));
}

Console.WriteLine($"Round 1: {counter1}"); // 55386
Console.WriteLine($"Round 2: {counter2}"); // 54824
